alacritty/src/display/hint.rs

```diff
@@ -259,30 +342,65 @@ pub fn highlighted_at<T>(
             return None;
         }
 
-        hint.regex.with_compiled(|regex| {
-            // Setup search boundaries.
-            let mut start = term.line_search_left(point);
-            start.line = max(start.line, point.line - MAX_SEARCH_LINES);
-            let mut end = term.line_search_right(point);
-            end.line = min(end.line, point.line + MAX_SEARCH_LINES);
+        // TODO: Use `bool::then` instead of double-if when MSRV >= 1.50.0
```

Comment:
```suggestion
        // TODO: Use `bool::then` instead of double-if with Rust version (MSRV) >= 1.50.0.
```

Comment should include `Rust` and `version` substrings to make it easier to find.